src/analyses/constant_propagator.cpp

```diff
@@ -504,6 +501,21 @@ bool constant_propagator_domaint::merge(
   return values.merge(other.values);
 }
 
+/// Attempt to evaluate expression using domain knowledge
+/// This function changes the expression that is passed into it.
+/// \param expr The expression to evaluate
+/// \param ns The namespace for symbols in the expression
+/// \return True if the expression is unchanged, false otherwise
+bool constant_propagator_domaint::try_evaluate(
+  exprt &expr,
+  const namespacet &ns) const
+{
+  adjust_float_expressions(expr, ns);
```

Comment:
Isn't it very inefficient to do this every time? Couldn't this instead be done once (as cbmc, goto-diff, jbmc do)?

Reply:
I'm not sure what you mean by that. Are you referring to `adjust_float_expressions`? Are you saying we should call adjust_float_expressions over all expressions in the program before doing anything else?

Reply:
My apologies for having been too brief: yes, that's what I'd suggest. The place you are doing it in right now might result in repeated applications of `adjust_float_expressions` to the very same expression multiple times.

Reply:
I'm actually not sure where the right place to do this would be in this case, do you have something specific in mind? Right after `initialize_goto_model`?